### DISCUSSION TITLE
What is the purpose of your .tt files?

### DISCUSSION THREAD

**Discussion by jeffward01** [xavierjohn/Trellis#15] [General]
Very cool library! I have been following you for a while over at https://github.com/vkhorikov/CSharpFunctionalExtensions 

I **very very** much appreciate your use of `xxxAsync` methods, the compiler I found would have issues with it, and also, it was hard to mix nested `sync` and `async` methods together implicitly, rather than explicitly.

My question is:

* In your code base, what is the purpose of your `.tt` and `.t4` files? I know that they are for code-generation, but how are they used?

Thanks!

**Comment by xavierjohn** [comment]
The template (T4) is used to generate this code
https://github.com/xavierjohn/FunctionalDDD/blob/main/Core/src/Functional/Result/Extensions/BindTs.cs

If you use Visual Studio, T4 to cs file is generated when the t4 file is saved.

**Reply by jeffward01** [reply]
Ah makes perfect sense, would not be fun to write that for each method implementation type.

Thanks for educating me!

PS: I was a long time VS user, love love VS, but it was 'heavy', so I use Rider now. But for some more exotic things like Incremental Source Generator development, some types of unit testing, and apprently `.t4` files also, VS is the better option.  I use VS in select scenarios like this. 

Unpopular opinion  - Vs Code is an abomination to C#  🙃

**Comment by jeffward01** [comment]
Is it possible to mark this as 'answered'? I don't see the option for it.

Thanks for the fastest answer ever! <3
